File: FRONTEND/lib/COMMON/Core/Models/Recommendation Model/hard_skills_model.dart
import 'package:origami_structure/imports.dart';

List<HardSkillsDataModel> hardSkillsDataListFromJson(String str) =>
    List<HardSkillsDataModel>.from(json.decode(str).map((x) => HardSkillsDataModel.fromJson(x)));

String hardSkillsDataListToJson(List<HardSkillsDataModel> data) =>
    json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

HardSkillsDataModel hardSkillsDataFromJson(String str) => HardSkillsDataModel.fromJson(json.decode(str));

String hardSkillsDataToJson(HardSkillsDataModel data) => json.encode(data.toJson());

class HardSkillsDataModel {
  HardSkillsDataModel({
    this.hardSkillID,
    this.jobField,
   this.jobSubField,
    this.jobSpecialization,
    this.jobTitle,
    this.typeOfSpecialization,
    this.hardSkillCategory,
    this.hardSkill,
    this.level,
  });

  String? hardSkillID;
  String? jobField;
  String? jobSubField;
  String? jobSpecialization;
  String? jobTitle;
  String? typeOfSpecialization;
  String? hardSkillCategory;
  String? hardSkill;
  String? level;

  factory HardSkillsDataModel.fromJson(Map<String, dynamic> json) =>
      HardSkillsDataModel(
        hardSkillID: json["hardSkillID"] ?? null,
        jobField: json["jobField"] ?? null,
        jobSubField: json["jobSubField"] ?? null,
        jobSpecialization: json["jobSpecialization"] ?? null,
        jobTitle: json["jobTitle"] ?? null,
        typeOfSpecialization: json["typeOfSpecialization"] ?? null,
        hardSkillCategory: json["hardSkillCategory"] ?? null,
        hardSkill: json["hardSkill"] ?? null,
        level: json["level"] ?? null,
      );

  Map<String, dynamic> toJson() =>
      {
        "hardSkillID": hardSkillID ?? null,
        "jobField": jobField ?? null,
        "jobSubField": jobSubField ?? null,
        "jobSpecialization": jobSpecialization ?? null,
        "jobTitle": jobTitle ?? null,
        "typeOfSpecialization": typeOfSpecialization ?? null,
        "hardSkillCategory": hardSkillCategory ?? null,
        "hardSkill": hardSkill ?? null,
        "level": level ?? null,
      };
}

File: BACKEND/Models/Recommendation Model/hard_skills_model.js
////////////////////////////////////////////// IMPORTS //////////////////////////////////////////////

const mongoose = require('mongoose');
const Schema = mongoose.Schema;

/////////////////////////////////////////// SCHEMA AND MODEL ///////////////////////////////////////////

// Create Skill Schema
const HardSkillsDataSchema = new Schema({
    hardSkillID: {
        type: String
    },
    jobField: {
        type: String
    },
    jobSubField: {
        type: String
    },
    jobSpecialization: {
        type: String
    },
    jobTitle: {
        type: String
    },
    typeOfSpecialization: {
        type: String
    },
    hardSkillCategory: {
        type: String
    },
    hardSkill: {
        type: String
    },
    level: {
        type: String
    },
})


var HardSkillsModel = mongoose.model('hard_skills_data', HardSkillsDataSchema, 'hard_skills_data');
module.exports = {
    HardSkillsDataSchema: HardSkillsDataSchema, 
    HardSkillsModel: HardSkillsModel
};